### ISSUE TITLE
Missing Include directories for CMake

### ISSUE THREAD

**Issue by mojovski** [cppan/cppan#18]
Hi, 
I feel somehow that I did not quite get the concept behind cppan: I would like to use boost in my project and have this cppan.yml file:
```
local_settings:
    cppan_dir: cppan

dependencies:
  pvt.cppan.demo.boost.system: 1
  pvt.cppan.demo.boost.coroutine: 1
  pvt.cppan.demo.boost.context: 1 
  pvt.cppan.demo.boost.thread: 1
  pvt.cppan.demo.openssl.crypto: 1
  pvt.cppan.demo.openssl.ssl: 1
```

In my CMakeLists.txt however, I follow the examples
but the examples do not show how I can access the include_dirs from the included libraries.
My cmake sofar:
```
cmake_minimum_required (VERSION 3.2)
project (Simple-WebSocket-Server)
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall")


#if windows is used, use cppan. the dependencies are declared in cppan.yml
if (WIN32)
	MESSAGE(STATUS "Entering Win32 Mode")
	add_subdirectory(cppan)
	SET(DEP_LIBS ${cppan})
	MESSAGE(STATUS "DEP_LIBS: ${DEP_LIBS}")

ELSE()
	# something else, which works fine on Linux....
ENDIF()

find_package(Threads REQUIRED)

#!!!!!!!!!!!!!! What to put here???????????????
include_directories(.)

add_executable(ws_examples ws_examples.cpp)
target_link_libraries(ws_examples ${CMAKE_THREAD_LIBS_INIT} ${DEP_LIBS})


if(MSYS)
    target_link_libraries(ws_examples ws2_32 wsock32)
    target_link_libraries(wss_examples ws2_32 wsock32)
endif()

enable_testing()
add_subdirectory(tests)
```

Later, when compiling the the executables, I of course, receive an error that the headers, included by
```c++
#include <openssl/env.h>
```
are not found.

Looking in to my include dirs inside the generated VS2015 project, there is only the "." directory inside.

I also tried to inspect some files generated inside the directory cppan after calling 
```
cppan
```
but had no luck. 

So, in short: How can I access the ${LIBNAME_INCLUDE_DIRS} from my main CMakeLists?

Thank you very much!

PS: The line in CMake
```
MESSAGE(STATUS "DEP_LIBS: ${DEP_LIBS}")
```
shows that ${DEP_LIBS} (=cppan) is empty.

**Comment by egorpugin** [comment]
Hi!

Try explicitly specify deps like this:
```
add_executable(ws_examples ws_examples.cpp)
target_link_libraries(ws_examples ${CMAKE_THREAD_LIBS_INIT} ${DEP_LIBS} 
  pvt.cppan.demo.boost.system
  pvt.cppan.demo.boost.coroutine
  pvt.cppan.demo.boost.context
  pvt.cppan.demo.boost.thread
  pvt.cppan.demo.openssl.crypto
  pvt.cppan.demo.openssl.ssl
)
```

**Comment by egorpugin** [comment]
There's also another interesting solution. :)
See this file https://github.com/cppan/cppan/blob/master/cmake/FindCPPAN.cmake
You could download it to your cmake modules and write in CMakeLists.txt.
```
find_package(CPPAN REQUIRED)
cppan_add_dependency(pvt.cppan.demo.boost.system)
...
cppan_execute()

...
target_link_libraries(yourtgt
  pvt.cppan.demo.boost.system
  ...
)
```

**Comment by mojovski** [comment]
Wow, thats magic! :)
I just tried your first proposition and it worked! Actually, I still dont get how the method target_link_libraries also access the include_directories but its fine.

Thank you also very much for the new feature preview. It sounds great!
